src/state_machine.rs

```diff
@@ -0,0 +1,1732 @@
+use hashbrown::HashMap;
+use p256k1::{ecdsa, point::Point};
+
+use crate::{common::Signature, taproot::SchnorrProof};
+
+/// A generic state machine
+pub trait StateMachine<S, E> {
+    /// Attempt to move the state machine to a new state
+    fn move_to(&mut self, state: S) -> Result<(), E>;
+    /// Check if the state machine can move to a new state
+    fn can_move_to(&self, state: &S) -> Result<(), E>;
+}
+
+/// Result of a DKG or sign operation
+#[allow(dead_code)]
```

Comment:
Do we need this #[allow(dead_code]? It seems like we use this enum? Or is it because we don't use SignTaproot yet?

Reply:
Yeah it was just bad copypasta.